python/lsst/daf/persistence/butler.py

```diff
@@ -634,9 +642,16 @@ def put(self, obj, datasetType, dataId={}, doBackup=False, **rest):
         for repoData in self._repos.outputs():
             location = repoData.repo.map(datasetType, dataId, write=True)
             if location:
-                if doBackup:
-                    repoData.repo.backup(datasetType, dataId)
-                repoData.repo.write(location, obj)
+                if isinstance(location, ButlerComposite):
+                    components = {}
+                    location.disassembler(obj=obj, dataId=location.dataId, componentDict=components)
```

Comment:
Might be nice to prefill `components` with the names of the datasets expected and even to pass the disassembler the expected dataset types.  It's also a little confusing to have a local `components` dict as well as `location.components`.